### PR TITLE
Create organizational structure to the code (via controllers)

### PR DESCRIPTION
**Pull request by ZainRizvi**

### CODE REVIEW

**Comment by ZainRizvi** [comment]
#approve 6c427da7b4352d7060fc96f1e9152137973405594fdc8c1b8e04f3999eeabcba9aaea965286847e429818c50682f426c3f1729a95bb1ba27ec5ac432669c9a9a1216d41825b9117b638c842d043f112796b382c6862bc34219ee5aa47e8888e6a2b75e4ed89241aa0902fa7a53239f8ed2f308b7761e00c15c70a81f756b7fc9075fe6143e942814a960dad7cc696f5c724b100060eb69cf218446374de540acf256519723b51f888e06d75a6200c61a2e8450850c0bf119f8c74a9ce3ac428e517c91df6a27d43008f60d4723b7c0feb74193c6a5c1ae5f27ac6565ac5d6625732ee00da140f648359b66cd9166321b8b00728b11ec0a98b10cc31457090ed4

**Comment by enomic** [comment]
Output is here: https://gist.github.com/c7fec37a07573f13a853

**Comment by ZainRizvi** [comment]
#approve 696b2dd2f69578a4a51f453d49ff9b2434490621ffdd09378653d7a5936d0c6567a68f43857bfd4ba78dbaef83e986c14fd0dd1abcff4da3cd3fe41e5140d91d7039f92fd67829b9bff47d161721012daa3f91e07ed7b86c1189576411b36e191d7e5f2da72ead6024f837926396aecac06aee6e2105f90a15e8cf1abbc273326067ad142d246c4cfdda822ae9a9082ec1aca7b1fcfbc8343642e19d54fd4a6284c8ea48500433ff1825b7eb6b861ff2ef058c0c314d4d58c2d0d88cefe57ef1c83bcede970ad2250463b15901cc8698edb0acf9e454c243893c5cdcae6c4c5969608e4d3ecbe6d98bece16dfd7a9357864f8378b42252eafd6a466852e45432

**Comment by enomic** [comment]
Output is here: https://gist.github.com/b61ce87435d5d10abf21

**Comment by ZainRizvi** [comment]
#approve 6c427da7b4352d7060fc96f1e9152137973405594fdc8c1b8e04f3999eeabcba9aaea965286847e429818c50682f426c3f1729a95bb1ba27ec5ac432669c9a9a1216d41825b9117b638c842d043f112796b382c6862bc34219ee5aa47e8888e6a2b75e4ed89241aa0902fa7a53239f8ed2f308b7761e00c15c70a81f756b7fc9075fe6143e942814a960dad7cc696f5c724b100060eb69cf218446374de540acf256519723b51f888e06d75a6200c61a2e8450850c0bf119f8c74a9ce3ac428e517c91df6a27d43008f60d4723b7c0feb74193c6a5c1ae5f27ac6565ac5d6625732ee00da140f648359b66cd9166321b8b00728b11ec0a98b10cc31457090ed4

**Comment by enomic** [comment]
Output is here: https://gist.github.com/568ca1febaf81814e99e

File: verify.js [hunk 1]
```diff
@@ -5,16 +5,18 @@ var path = require('path');
 var FS_ENCODING = {encoding: 'utf8'};
 
 function verify(commitId, signature, logger) {
-  var verifier = crypto.createVerify('RSA-SHA256');
-  verifier.update(commitId);
   for (var i = 0; i < adminIds.length; i++) {
+    var verifier = crypto.createVerify('RSA-SHA256');
```
Comment (enomic) [inline verify.js:9]: Oops.. I suspected this was the problem! Thanks for fixing it :-D

File: controllers/index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// controllers/index.js
+
+(function (controllers) {
+
+    var githubController = require("./githubController");
+    
+    controllers.init = function (app) {
```
Comment (enomic) [inline controllers/index.js:7]: What calls this? I don't think the code will work in the current form

Comment (ZainRizvi) [inline controllers/index.js:7]: It's called from server.js (unless I screwed up the merge...)

Sent from my Windows Phone

-----Original Message-----

Comment (enomic) [inline controllers/index.js:7]: The fix worked! Now you should be able to approve code, including this once you rebase

**Comment by enomic** [comment]
Actually, verification with that fix is still broken.

**Comment by ZainRizvi** [comment]
You were right, the controller was broken. I had borked the merge :P

It's fixed now

**Comment by ZainRizvi** [comment]
#approve 34e7c3a4bcf09cd877fa66395138260cb9116d19b20862cbd8b0cbab40dacd630a7e93c63b32d2a22345f36f90ee035771c4d035085626d1c5b2e7829471da810195317dae04e54cbdee10c3e4fb5e095abaacdb28efa4ca354cb13fa590dc52c7559dea5c2ae059f25807dbe7703e1c9213f62a26ed21877d2ee04a056cd1a4e8e2103f47f0751cf15ab3b81138962502c8d034edd8804975c310883e12cc94e7be4ae5fabc2378a686392ff582da8c23afbe1d6971c0bd46a6f25348fc159d4a5d224868278713458f95cb88e6e1594d1f098886528b8c3c7e3b294ea48b8d3d9a316854f9e316e8d41ff558df6d256737e6a4f08a6dfa20457315712d8fdf

**Comment by enomic** [comment]
Output is here: https://gist.github.com/821ddc5f2e4c27f41061

**Pull request merged**